src/Microsoft.Identity.Client/ClientAssertion.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Type containing an assertion representing a clients's credentials. This type is used to allow specific claims to be added to the authentication request.<c>UserAssertion</c>
+    /// See https://aka.ms/msal-net-client-credentials
+    /// </summary>
```

Comment:
@jmprieur what other details should I provide here?

Reply:
This is not client credentials. this is a clientassertion. You should though generally be able to copy the text from ADAL?

---
In reply to: [291385278](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1208#discussion_r291385278) [](ancestors = 291385278)

Reply:
I thought a client assertion was a type of credential? This is a quote from our [wiki](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Client-credential-flows#overview-of-msalnet-public-api-for-client-credentials)
`Overview of MSAL.NET public API for client credentials`
`MSAL.NET supports three types of client credentials:`

`Application secrets`
`Certificates`
`Optimized client assertions`

Reply:
MSAL Wiki needs updating to be more precise. The clientassertion is not credentials

---
In reply to: [291444583](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1208#discussion_r291444583) [](ancestors = 291444583)

Reply:
Wiki and comments will be updated early next week